core/delegation/delegation.go

```diff
@@ -145,6 +135,63 @@ func NewDelegationView(root ipld.Link, bs blockstore.BlockReader) (Delegation, e
 	return NewDelegation(blk, bs)
 }
 
+// Export the blocks that comprise the delegation, including all extra attached
+// blocks.
+func Export(rt ipld.Block, blks blockstore.BlockReader, atchblks blockstore.BlockReader) iter.Seq2[ipld.Block, error] {
+	return func(yield func(ipld.Block, error) bool) {
+		ucan, err := Decode(rt)
```

Comment:
calling `Blocks` on a delegation will call `Export` on that delegation, which has already the root block decoded into a `ucan.View` on creation. A way to avoid decoding it here again would be to have the `Export` function take a `ucan.View` instead of a root block.

I'm assuming here that decoding the block is an expensive operation, given that it was done lazily in the original implementation. Please disregard this comment if that's not the case.